### ISSUE TITLE
Add other-extensions field

### ISSUE THREAD

**Issue by phadej** [bos/aeson#472]
I'll do it at some point, if nobody else is quicker

**Comment by hvr** [comment]
Be careful about `TemplateHaskellQuotes` vs `TemplateHaskell` though (see also  https://github.com/bos/aeson/pull/347#commitcomment-15726665)

**Comment by bergmark** [comment]
This seems like something that should be automated, it's not likely that we'll remember to keep this list up to date.

**Comment by seagreen** [comment]
https://github.com/haskell/cabal/issues/1562#issuecomment-243627994

**Comment by phadej** [comment]
FWIW, I'm on-off working on https://github.com/phadej/cabal-refact, inferring other-extensions is on todo list

**Comment by seagreen** [comment]
@phadej: That's awesome! Will it work in the presence of CPP? I can't find the comment, but when I originally suggested [removing other-extensions from .cabal files](https://github.com/haskell/cabal/issues/3081) I was helpfully told that they aren't always machine-derivable due to CPP.

**Comment by phadej** [comment]
@seagreen I run `cpphs` with different `__GLASGOW_HASKELL__` (and without) and take the intersection of pragmas. CPP guarded pragmas aren't interesting.

Also with `cabal-refact` there's "freedom" to not be always correct, as the changes should be reviewed by the human, so if there's some scenario where above won't work, it's not a show-stopper.

**Comment by seagreen** [comment]
> Also with cabal-refact there's "freedom" to not be always correct, as the changes should be reviewed by the human, so if there's some scenario where above won't work, it's not a show-stopper.

Gotchya. Well regardless of whether this eventually infer `other-extensions` well enough for it to be not needed in `.cabal` files, it's still a great project!

**Comment by phadej** [comment]
@seagreen FWIW, it will be needed for at least as long as we can (or cannot) infer dependency bounds. So won't hold my breath. :( Short reasoning: tooling needs that information (i.e. to not try to install package it cannot), but it's too expensive to fetch the source tarball and look into it (during the dependency solving), so it's need be ready available somewhere. The `.cabal` file is obvious. And differently from version bounds, the `other-extension` doesn't even change. So the first step, where `cabal-refact` is integrated into CI would be good enough!

**Comment by seagreen** [comment]
@phadej: I agree with most of that. Going to email you about the rest since I've taken us way off topic from `aeson`:)

**Comment by bergmark** [comment]
How has the progress been on cabal-refact @phadej?

**Comment by phadej** [comment]
@bergmark slowly; have to find time & motivation to work on boring bits
